src/xunit.performance.api/ETWProfiler.cs

```diff
@@ -107,9 +119,196 @@ public static void Profile(string assemblyFileName, string sessionName, Action a
+
+                using (var context = new EtwPerformanceMetricEvaluationContext(
+                    fileName, source, performanceTestMessages, sessionName))
+                {
+                    source.Process();
+                    return context;
+                }
+            }
+        }
+
+        private static void WriteXmlFile(string xmlFileName, AssemblyModel assemblyModel)
+        {
+            var xmlAssembliesElement = new XElement("assemblies");
+            var xmlAssemblyElement = new XElement("assembly", new XAttribute("name", Path.GetFileName(assemblyModel.Name)));
+            xmlAssembliesElement.Add(xmlAssemblyElement);
+            var xmlCollectionElement = new XElement("collection");
+            xmlAssemblyElement.Add(xmlCollectionElement);
+
+            foreach (var testModel in assemblyModel.Collection)
+            {
+                var xmlTestElement = new XElement(
+                    "test",
+                    new XAttribute("name", testModel.Name),
+                    new XAttribute("type", testModel.ClassName),
+                    new XAttribute("method", testModel.Method));
+                xmlCollectionElement.Add(xmlTestElement);
+                var xmlPerformanceElement = new XElement("performance");
+                xmlTestElement.Add(xmlPerformanceElement);
+                var xmlMetricsElement = new XElement("metrics");
+                xmlPerformanceElement.Add(xmlMetricsElement);
+
+                foreach (var metric in testModel.Performance.Metrics)
+                {
+                    var xmlMetricElement = new XElement(
+                        metric.Name,
+                        new XAttribute("displayName", metric.DisplayName),
+                        new XAttribute("unit", metric.Unit));
+                    xmlMetricsElement.Add(xmlMetricElement);
+                }
+
+                var xmlIterationsElement = new XElement("iterations");
+                xmlPerformanceElement.Add(xmlIterationsElement);
+
+                var index = 0;
+                foreach (var iterationModel in testModel.Performance.IterationModels)
+                {
+                    var xmlAttributes = new List<XAttribute>
+                    {
+                        new XAttribute("index", index++)
+                    };
+
+                    foreach (var kvp in iterationModel.Iteration)
+                    {
+                        xmlAttributes.Add(new XAttribute(kvp.Key, kvp.Value));
+                    }
+
+                    var xmlIterationElement = new XElement(
+                        "iteration",
+                        xmlAttributes.ToArray());
+                    xmlIterationsElement.Add(xmlIterationElement);
+                }
+            }
+
+            var xmlDoc = new XDocument(xmlAssembliesElement);
+            using (var xmlFile = File.Create(xmlFileName))
+            {
+                xmlDoc.Save(xmlFile, SaveOptions.DisableFormatting);
```

Comment:
How much space do we save by disabling formatting? I'm curious whether the tradeoff is worth it (since we lose "readability" when debugging issues). Sadly the files can get long enough to defeat most auto-formatters...